Moved osg-flock docs to IDTOKEN, added system requirements.  (SOFTWARE-4548) (#804)
opensciencegrid/docs: docs/submit/osg-flock.md

In other words, you do not need a whole HTCondor batch system just to have a local OSG access point.

System Requirements
-------------------

> The hardware requirement for an OSG access point depends on several factors such as number of users, 
> number of jobs and for example how I/O intensive those jobs are. Our minimum recommended configuration
> is 6 cores, 12 GB RAM and 1 TB of local disk. The hardware can be bare metal or virtual machine, but
> we do not recommend containers as these submit host are running a bunch of system services which can
> be difficult to configure in a container.

Reviewer: ```suggestion
The hardware requirement for an OSG access point depends on several factors such as number of users,
number of jobs and for example how I/O intensity of those jobs.
Our minimum recommended configuration is 6 cores, 12 GB RAM and 1 TB of local disk.
The hardware can be bare metal or virtual machine, but we do not recommend containers as these submit host are running
many system services which can be difficult to configure in a container.
```

Minor wording changes and line splits to try to reduce future diffs: https://opensciencegrid.org/technology/documentation/style-guide/#meta

is 6 cores, 12 GB RAM and 1 TB of local disk. The hardware can be bare metal or virtual machine, but
we do not recommend containers as these submit host are running a bunch of system services which can
be difficult to configure in a container.

Also consider the following technical requirements:
Also consider the following configuration requirements:

* __Operating system:__ A RHEL 7 compatible operating system.
* __User IDs:__ If it does not exist already, the installation will create the Linux user ID `condor`.
* __Host certificate:__ Recommended for authentication to OSG infrastructure.
  See our [documentation](../security/host-certs.md) for instructions on how to request and install host certificates.
* __Network:__ 
    * Inbound TCP port 9618 must be open.
    * The access point must have a public IP address with both forward and reverse DNS configured.

As with all OSG software installations, there are some one-time steps to prepare in advance:

* Obtain root access to the host
* Prepare [the required Yum repositories](../common/yum.md)
* Install [CA certificates](../common/ca.md)

> 

Reviewer: I'd prefer to keep the first two bullets here. For the third bullet, is `osg-flock` configured to use system CAs and do we not need to worry about IGTF CAs?

Author: We do not need CAs anymore - these are just straight tokens. The Yum repos install is in the next paragraph.

Reviewer: IIRC there was one last place that GSI was used in the GlideinWMS inf. Was that between factory and FE?

Author: That is only FE->factory->glidein. No GSI needed between submit hosts and CM.

Follow the [general registration instructions](../common/registration.md#new-resources).
For historical reasons, the service type is `Submit Node`.

### Request to be allowed to flock to OSG
OSG staff will need to add your access point to the list of hosts that flocked jobs are accepted from,
and provide a HTCondor IDTOKEN so that your new access point can authenticate with OSG.
Send email to <help@opensciencegrid.org> with the hostname of your access point and request to be added to the list.

Installing Required Software
----------------------------
Flocking requires HTCondor software as well as software for reporting to the OSG accounting system.
Start by setting up the OSG YUM repositories following the
[Installing Yum Repositories](../common/yum.md). __Note that you have to use OSG 3.6__. Earlier

> versions will not work.

Reviewer: Why is this the case? We ship 8.9.11 in 3.5 upcoming. I'm a little hesitant to push 3.6 right now but as long as we know that this works, I'm ok with telling folks to use 3.6

Author: 3.6 is better long term here. That is what I did the testing with.

Reviewer: Ok sure. Could you make this note an [admonition](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types) to highlight it further?

Author: Yes, but let's get this in place and I can take a second pass on it later

Configuring Authentication
--------------------------

> Create a file named `/etc/condor/tokens.d/flock.opensciencegrid.org` with the IDTOKEN you
> received earlier. The token you only take up one line (that is, there should not be any
> line breaks in the token).

Reviewer: ```suggestion
Create a file named `/etc/condor/tokens.d/flock.opensciencegrid.org` with the IDTOKEN you received earlier.
Ensure that there aren't any line breaks in this file (i.e., the entire token should only take up one line).
```  

Wording/spacing suggestion
